Quelque chose m'échappe. 

S'il était pour moi logique d'incrémenter de 1 le paramètre du test, en lançant la commande de test en local (`python manage.py test zds.gallery`), celui-ci échouait lamentablement avec le message `AssertionError 3 != 4`.

Un résultat qui m'a étonné compte tenu du message d'erreur de Travis ([build #5281.1](https://travis-ci.org/zestedesavoir/zds-site/jobs/115506272#L911)). Néanmoins, je me suis dit qu'étant donné que j'intégrais deux nouveaux filtres, il fallait incrémenter le paramètre de 2. Après l'avoir fait, tous les tests passent en local.

```
$ python manage.py test zds.gallery
Creating test database for alias 'default'...
..................................................................
----------------------------------------------------------------------
Ran 66 tests in 9.238s

OK
Destroying test database for alias 'default'...
```

Mais dès que je _push_, Travis n'obtient pas le même résultat que moi :

```
FAIL: test_success_new_image_write_permission (zds.gallery.tests.tests_views.NewImageViewTest)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/zestedesavoir/zds-site/zds/gallery/tests/tests_views.py", line 636, in test_success_new_image_write_permission

    self.assertEqual(4, len(os.listdir(self.gallery.get_gallery_path())))  # New image and thumbnail

AssertionError: 4 != 3
```

Et là ... **Pourquoi ?** D'où vient cette différence de résultat ? Quelqu'un aurait une idée/explication ?

--- a/templates/gallery/image/edit.html
+++ b/templates/gallery/image/edit.html
@@ -1,6 +1,7 @@
 {% extends "gallery/base.html" %}
 {% load crispy_forms_tags %}
 {% load thumbnail %}
+{% load remove_url_protocole %}
 {% load i18n %}
 
 
@@ -30,12 +31,23 @@
         <p>
             {% trans "Image" %} :
         </p>
-        <a href="{{ image.physical.url }}"><img src="{{ image.physical.url }}" alt="{{ image.legend }}"></a>
-        
+        <a href="{{ image.physical.url|remove_url_protocole }}">
+            <img src="{{ image.physical.gallery_illu.url|remove_url_protocole }}" alt="{{ image.legend }}">
+        </a>
+
         <p>
             {% trans "Miniature" %} :
         </p>
-        <a href="{{ image.physical.gallery.url }}"><img src="{{ image.physical.gallery.url }}" alt="{{ image.legend }}"></a>
+        <a href="{{ image.physical.gallery.url|remove_url_protocole }}">
+            <img src="{{ image.physical.gallery.url|remove_url_protocole }}" alt="{{ image.legend }}">
+        </a>
+
+        {% if perms.featured.change_featuredresource %}
+        <p>
+            {% trans "Lien pour utiliser cette image en une" %} : <br>
+            <input type="text" value="{{ app.site.url }}{{ image.physical.featured.url }}" readonly onclick="this.select()">
+        </p>
+        {% endif %}
     </div>
     <div class="gallery-col-edit">
         {% if gallery_mode.can_write %}
@@ -55,7 +67,7 @@
             {% trans "Miniature + lien vers taille normale" %} :
             <input type="text" value="[![{{ image.legend }}]({{app.site.url}}{{ image.physical.gallery.url }})]({{app.site.url}}{{ image.physical.url }})" readonly onclick="this.select()">
         </p>
-        
+
         {% crispy as_avatar_form %}
     </div>
 {% endblock %}

--- a/zds/gallery/tests/tests_views.py
+++ b/zds/gallery/tests/tests_views.py
@@ -459,7 +459,7 @@ class EditImageViewTest(TestCase):
                 follow=True
             )
         self.assertEqual(200, response.status_code)
-        self.assertEqual(nb_files + 2, len(os.listdir(self.gallery.get_gallery_path())))
+        self.assertEqual(nb_files + 3, len(os.listdir(self.gallery.get_gallery_path())))
 
         image_test = Image.objects.get(pk=self.image.pk)
         self.assertEqual('edit title', image_test.title)
@@ -633,7 +633,7 @@ class NewImageViewTest(TestCase):
 
         self.assertEqual(200, response.status_code)
         self.assertEqual(1, len(self.gallery.get_images()))
-        self.assertEqual(2, len(os.listdir(self.gallery.get_gallery_path())))  # New image and thumbnail
+        self.assertEqual(3, len(os.listdir(self.gallery.get_gallery_path())))  # New image and thumbnail
         self.gallery.get_images()[0].delete()
 
     def test_fail_new_image_with_read_permission(self):

--- a/zds/settings.py
+++ b/zds/settings.py
@@ -204,6 +204,8 @@ THUMBNAIL_ALIASES = {
         'help_illu': {'size': (48, 48), 'crop': True},
         'help_mini_illu': {'size': (26, 26), 'crop': True},
         'gallery': {'size': (120, 120), 'crop': True},
+        'featured': {'size': (228, 228), 'crop': True},
+        'gallery_illu': {'size': (480, 270), 'crop': True},
         'content': {'size': (960, 960), 'crop': False},
     },
 }
@@ -568,4 +570,3 @@ try:
     from settings_prod import *
 except ImportError:
     pass
-

